pandas/core/indexes/api.py

```diff
@@ -213,14 +215,41 @@ def conv(i):
 
     if kind == "special":
         result = indexes[0]
+        first = result
+
+        dtis = [x for x in indexes if isinstance(x, DatetimeIndex)]
+        dti_tzs = [x for x in dtis if x.tz is not None]
+        if len(dti_tzs) not in [0, len(dtis)]:
```

Comment:
can we not just kill union_many now (and just have it call union_indexes)?

Reply:
that would change the behavior of union_many, which in principle would require a deprecation

Reply:
ok, but we can just deprecate this anyhow (union_many)? as you have seemed to replaced it

Reply:
to be honest i am ok with just killing it and replacing the code path. its annoying keeping these 2 different paths.

Reply:
This _is_ deprecating union_many

Reply:
ok yeah i guess deprecation is ok